src/misc/rwqueue.cpp

```diff
@@ -74,6 +74,53 @@ void RWQueue<T>::Enqueue(T&& item)
 	has_items.notify_one();
 }
 
+template <typename T>
+void RWQueue<T>::BulkEnqueue(std::vector<T>& from_source, const size_t num_requested)
+{
+	// Check that the source can provide the quantity requested
+	assert(num_requested <= from_source.size());
+
+	// The source iterator will be incremented with each chunk enqueued
+	auto source_start = from_source.begin();
+
+	// The remaining count is decremented until no more items need enqueing
+	auto num_remaining = num_requested;
+
+	while (num_remaining > 0) {
+		// Begin exclusive access to the internal queue
+		std::unique_lock<std::mutex> lock(mutex);
+
+		// Limit the quantity to our capacity
+		const auto min_items = std::max(static_cast<size_t>(1),
+		                                static_cast<size_t>(capacity - queue.size()));
+		const auto num_items = std::min(num_remaining, min_items);
+
+		// Set the range of the source items we'll be working with
+		const auto stride     = static_cast<stride_t>(num_items);
```

Comment:
Hmm, is this really a "stride" type of value? Is it not just a simple offset, and therefore should be called that?

Reply:
Actually, maybe get rid of this temp var altogether? It's only used on the below live. The word "stride" makes me expect all sorts of things that are just not happening here 😄

Reply:
Good call - indeed, it's just an interator "difference type" (which probably is a size_t).  But I was getting warnings adding it as is.

Renamed to `difference_t`, as a slightly less nasty form of the vector's iterator `difference_type`:

``` c++
using difference_t = typename std::vector<T>::difference_type;
```